modin/core/storage_formats/base/query_compiler.py

```diff
@@ -197,7 +208,25 @@ def default_to_pandas(self, pandas_op, *args, **kwargs) -> Self:
     # some of these abstract methods, but for the sake of generality they are
     # treated differently.
 
-    lazy_execution = False
+    lazy_row_labels = False
+    lazy_row_count = False
+    lazy_column_types = False
+    lazy_column_labels = False
+    lazy_column_count = False
+
+    @property
+    def lazy_shape(self):
```

Comment:
I see where this is used below, but is the check for both `lazy_row_count` and `lazy_column_count` necessary in both cases?

Reply:
I think so, at least to preserve the previous behavior for the time being. This flag usually guards a call to `DataFrame.empty`, which is defined to check both axes, and I'm worried about any breaking changes that might arise from only checking one dimension.